src/System.Data.SqlClient/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -1071,5 +1071,42 @@ static internal Version GetAssemblyVersion()
 
             return s_systemDataVersion;
         }
+
+
+        static internal readonly string[] AzureSqlServerEndpoints = {Res.GetString(Res.AZURESQL_GenericEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_GermanEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_UsGovEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_ChinaEndpoint) };
+
+        static internal bool IsAzureSqlServerEndpoint(string dataSource)
+        {
+            // remove server port
+            var i = dataSource.LastIndexOf(',');
```

Comment:
This code was copied from closed source code for opening purpose, and I thought I should keep it as it is. Should it be changed?